cmake/kconfig.cmake

```diff
@@ -90,6 +90,8 @@ endforeach()
 # A list of common environment settings used when invoking Kconfig during CMake
 # configure time or menuconfig and related build target.
 string(REPLACE ";" "\\\;" SHIELD_AS_LIST_ESCAPED "${SHIELD_AS_LIST}")
+# cmake commands are escaped differently
+string(REPLACE ";" "\\;" SHIELD_AS_LIST_ESCAPED_COMMAND "${SHIELD_AS_LIST}")
```

Comment:
instead of adding one more escape list variant then maybe update the place where this is used to expect `\;` for list splitting, instead of just `;`.

That would be changing this code:
https://github.com/zephyrproject-rtos/zephyr/blob/2cf91050c88ed65f71b24beac155c36adacb43c3/scripts/kconfig/kconfigfunctions.py#L487
into:
```
 return "y" if shield in list.split("\;") else "n" 
```

Reply:
Forget this proposal for now, it might impact the places where `add_custom_target()` is used.